### PR TITLE
ApplicationPlayer: Fix playlists by correctly switching players

### PR DESCRIPTION
**Pull request by fritsch**
Fixes mixed playlists. In that case we need to switch from e.g. paplayer to videoplayer or another player. Fix was posted by Fernet on slack.

### CODE REVIEW

**Comment by fritsch** [comment]
@DaveTBlake please test and report back

File: xbmc/ApplicationPlayer.cpp [hunk 1]
```diff
@@ -115,6 +115,15 @@ bool CApplicationPlayer::OpenFile(const CFileItem& item, const CPlayerOptions& o
       return true;
     }
   }
+  else if (player && player->m_name != newPlayer)
+  {
+    CloseFile();
+    if (player->m_name != newPlayer)
```
Comment (ace20022) [inline xbmc/ApplicationPlayer.cpp]: Does CloseFile() change player->m_name ?

Comment (fritsch) [reply xbmc/ApplicationPlayer.cpp]: You mean, cause it is checked again?

Comment (fritsch) [reply xbmc/ApplicationPlayer.cpp]: In short: Neither PAPlayer nor VideoPlayer change the Player name, when CloseFile is called on them, but CloseFiles has a "reopen" argument ... 

I think we don't need this to be checked again. @FernetMenta what do you think?

**Comment by fritsch** [comment]
I tested it my self:
With Queue from menu, it won't work:
```
10:08:41.702 T:140129800808192   ERROR: CAudioDecoder: Unable to Init Codec while loading file /home/fritsch/Downloads/180709-Star-Wars.mp4
10:08:41.702 T:140129800808192 WARNING: PAPlayer::QueueNextFileEx - Failed to create the decoder
10:08:41.864 T:140131053496448  NOTICE: VideoPlayer::OpenFile: /home/fritsch/Downloads/MVI_7530.AVI
```

Still tries to use PAPlayer QueueNextFileEx to enqueue the AVI file which it should not do.

**Comment by fritsch** [comment]
With Queue the problem is that:

```
bool CApplicationPlayer::QueueNextFile(const CFileItem &file)
{
  std::shared_ptr<IPlayer> player = GetInternal();
  return (player && player->QueueNextFile(file));
}
``` 

gets called and no reopen happens.

Will try now mixed playlist, which should use the other path

**Comment by fritsch** [comment]
jenkins build this please

(Sorry for your double work)

**Pull request merged**